scripts/abins/input/dmol3loader.py

```diff
@@ -41,245 +55,233 @@ def read_vibrational_or_phonon_data(self):
+            data['unit_cell'] = self._read_lattice_vectors(dmol3_file)
 
             # read info about atoms and construct atom data
             masses = self._read_masses_from_file(obj_file=dmol3_file)
-            self._read_atomic_coordinates(file_obj=dmol3_file, data=data, masses_from_file=masses)
-            self._num_atoms = len(data['atoms'])
+            atoms = self._read_atomic_coordinates(file_obj=dmol3_file)
+            self.check_isotopes_substitution(atoms=atoms, masses=masses)
+            self._num_atoms = len(atoms)
+            data['atoms'] = atoms
+
+            # Update masses based on behaviour of check_isotopes_substitution
+            masses = [atom['mass'] for _, atom in atoms.items()]
 
             # read frequencies, corresponding atomic displacements and construct k-points data
-            self._parser.find_first(file_obj=dmol3_file, msg="Frequencies (cm-1) and normal modes ")
-            self._read_modes(file_obj=dmol3_file, data=data)
+            TextParser.find_first(file_obj=dmol3_file, msg="Frequencies (cm-1) and normal modes ")
+            frequencies, raw_modes = self._read_modes(dmol3_file)
+            data.update(self._assemble_mode_data(frequencies, raw_modes, masses))
 
             # save data to hdf file
             self.save_ab_initio_data(data=data)
 
             # return AbinsData object
             return self._rearrange_data(data=data)
 
-    def _read_lattice_vectors(self, obj_file=None, data=None):
+    @classmethod
+    def _read_lattice_vectors(cls, file_obj):
         """
         Reads lattice vectors from .outmol DMOL3 file.
         :param obj_file: file object from which we read
         :param data: Python dictionary to which found lattice vectors should be added
         """
-        pos = obj_file.tell()
-
-        try:
-            self._parser.find_first(file_obj=obj_file, msg="$cell vectors")
-
-        except EOFError:  # No unit cell for non-periodic calculations; set to zero
-            data["unit_cell"] = np.zeros(shape=(3, 3), dtype=FLOAT_TYPE)
-
-        else:
-            dim = 3
-            vectors = []
-            for i in range(dim):
-                line = obj_file.readline().split()
-                vector = [self._convert_to_angstroms(string=s) for s in line]
-                vectors.append(vector)
+        with TextParser.save_excursion(file_obj):
+            try:
+                TextParser.find_first(file_obj=file_obj, msg="$cell vectors")
 
-                data["unit_cell"] = np.asarray(vectors).astype(dtype=FLOAT_TYPE)
+            except EOFError:
+                # No unit cell for non-periodic calculations; set to zero
+                unit_cell = np.zeros(shape=(3, 3), dtype=FLOAT_TYPE)
 
-        obj_file.seek(pos)
+            else:
+                vectors = [file_obj.readline().split() for _ in range(3)]
+                unit_cell = np.asarray(vectors, dtype=FLOAT_TYPE)
+                unit_cell *= ATOMIC_LENGTH_2_ANGSTROM
```

Comment:
why 2 angstrums?

Reply:
I think 2 is a 1337 abbreviation for "TWO" in this case. Agree that it's confusing, could change it.

Technically that would be an API break if anyone is importing that from the Abins modules

Reply:
you could for this file do

`` 
import ATOMIC_LENGTH_2_ANGSTROM as BOHR_TO_ANGSTROM
```

or similar. Then it is not an API break, but in this code is more readable.

Reply:
Right, I much prefer referring to Bohr explicitly anyway. It might as well happen as an alias in the constants file, and there can be a note of the intended deprecation somewhere for when the version number is next bumped.